libblkid/src/superblocks/scoutfs.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2025 Versity, Inc.
+ *
+ * This file may be redistributed under the terms of the
+ * GNU Lesser General Public License.
+ */
+
+#include <inttypes.h>
+#include "superblocks.h"
+#include "crc32c.h"
+
+#define SCOUTFS_UUID_BYTES 16
+
+#define SCOUTFS_BLOCK_SM_SHIFT 12
+#define SCOUTFS_BLOCK_SM_SIZE  (1 << SCOUTFS_BLOCK_SM_SHIFT)
+#define SCOUTFS_BLOCK_LG_SHIFT 16
+#define SCOUTFS_BLOCK_LG_SIZE  (1 << SCOUTFS_BLOCK_LG_SHIFT)
+
+struct scoutfs_block_header {
+	uint32_t crc;
+	uint32_t magic;
+	uint64_t fsid;
+	uint64_t seq;
+	uint64_t blkno;
+};
+
+#define SCOUTFS_FLAG_IS_META_BDEV 0x01
+
+struct scoutfs_super_block {
+	struct scoutfs_block_header hdr;
+	uint64_t id;
+	uint64_t fmt_vers;
+	uint64_t flags;
+	uint8_t uuid[SCOUTFS_UUID_BYTES];
+	/* rest omitted */
+};
+
+static int probe_scoutfs(blkid_probe pr, const struct blkid_idmag *mag, int flag)
+{
+	const struct scoutfs_super_block *sb;
+	const unsigned char *buf = NULL;
+	uint32_t crc;
+	sb = blkid_probe_get_sb(pr, mag, struct scoutfs_super_block);
+	if (sb == NULL)
+		return errno ? -errno : 1;
+
+	/* scoutfs_super_block is always in a SCOUTFS_BLOCK_SM_SIZE block */
+	buf = blkid_probe_get_sb_buffer(pr, mag, SCOUTFS_BLOCK_SM_SIZE);
+	crc = crc32c(~0, (char *)buf + sizeof(uint32_t), SCOUTFS_BLOCK_SM_SIZE - sizeof(uint32_t));
+	if (!blkid_probe_verify_csum(pr, crc, le32_to_cpu(sb->hdr.crc)))
+		return BLKID_PROBE_NONE;
+
+	blkid_probe_sprintf_version(pr, "%"PRIu64, le64_to_cpu(sb->fmt_vers));
+
+	blkid_probe_set_uuid(pr, sb->uuid);
+
+	blkid_probe_sprintf_value(pr, "fsid", "%016"PRIx64, le64_to_cpu(sb->hdr.fsid));
+
+	blkid_probe_set_wiper(pr, 65536, 256);
+
+	if (flag) {
+		/* meta blocksize is 64k blocks */
+		blkid_probe_set_fsblocksize(pr, SCOUTFS_BLOCK_LG_SIZE);
+		blkid_probe_set_block_size(pr, SCOUTFS_BLOCK_LG_SIZE);
+
+		return le64_to_cpu(sb->flags) & SCOUTFS_FLAG_IS_META_BDEV ? 0 : BLKID_PROBE_NONE;
```

Comment:
These `return`s are unnecessarily complex.

Just do:

```c
if (!le64_to_cpu(sb->flags) & SCOUTFS_FLAG_IS_META_BDEV))
    return BLKID_PROBE_NONE;
```

Without the negation in the `else` block.
And `return 0` at the end.